service/storage_service.hh

```diff
@@ -761,6 +761,17 @@ private:
         // The node uses raft-based topology operations
         raft
     };
+    // The _topology_change_kind_enabled variable is first initialized in `join_cluster`.
+    // After the node successfully joins, the control over the variable is yielded
+    // to `topology_state_load`, so that it can control it during the upgrade from gossiper
+    // based topology to raft-based topology.
+    // FIXME: This boolean flag is mostly needed because, shortly after starting
+    // a new group 0, the state of `system.topology` is empty, and updating the
+    // `_topology_change_kind_enabled` variable from group 0 state would lead
+    // to wrong results. This could be fixed by writing an initial `system.topology`
```

Comment:
Do we even need `_topology_change_kind_enabled` in addition to `upgrade_state` in the topology? Why can't we have a function that returns `kind` depending on upgrade_state. It will also know how to handle empty topology.

Reply:
One issue is this FIXME. Another issue is that it affects gossip callbacks in storage_service, and gossiping starts before joining group 0 - we need these callbacks to operate in raft topology mode before we join a cluster that operates in raft topology mode.